xrootd/client/client.go

```diff
@@ -44,12 +47,39 @@ type Client struct {
 	protocolVersion  int32
 	signRequirements xrdproto.SignRequirements
 	seqID            int64
+	auths            map[string]auth.Auther
+}
+
+// Option configures an XRootD client.
+type Option func(*Client) error
+
+// WithAuth adds an authentication mechanism to the XRootD client.
+// If an authentication mechanism was already registered for that provider,
+// it will be silently replaced.
+func WithAuth(a auth.Auther) Option {
+	return func(client *Client) error {
+		return client.setAuth(a)
+	}
+}
+
+func (client *Client) setAuth(auth auth.Auther) error {
+	client.auths[auth.Provider()] = auth
+	return nil
+}
+
+func (client *Client) initSecurityProviders() {
+	providers := []auth.Auther{krb5.Default, unix.Default}
+	for _, provider := range providers {
+		if provider != nil {
+			client.auths[provider.Provider()] = provider
+		}
+	}
 }
 
 // NewClient creates a new xrootd client that connects to the given address using username.
 // When the context expires, a response handling is stopped, however, it is
 // necessary to call Cancel to correctly free resources.
-func NewClient(ctx context.Context, address string, username string) (*Client, error) {
+func NewClient(ctx context.Context, address string, username string, opts ...Option) (*Client, error) {
```

Comment:
please add a doc comment about this new optional `opts` argument. (probably before the blurb about the `context.Context` expiration.)